README.md

```diff
@@ -401,6 +434,44 @@ of clustering stanza in `etc/system/local/server.conf`).
 Indexer clustering is used to achieve some data availability & recovery. To learn
 more about Splunk indexer clustering, refer to [Splunk Docs](http://docs.splunk.com/Documentation/Splunk/latest/Indexer/Aboutclusters).
 
+## setup_shclustering
+
+This recipe sets up Splunk search head clustering. The attribute
+`node['splunk']['shclustering']['enabled']` must be set to true in order to
+run this recipe. Similar to `setup_auth`, this recipes loads
+the same encrypted data bag with the Splunk `shcluster_secret` key (to be shared among
+cluster members), using the [chef-vault cookbook](http://ckbk.it/chef-vault)
+helper method, `chef_vault_item`. See __Usage__ for how to set this up. The
+recipe will edit the cluster configuration, and then write a state file to
+`etc/.setup_shcluster` to indicate in future Chef runs that it has set the node's 
+search head clustering configuration. If cluster configuration should be changed, 
+then that file should be removed.
+
+It will also search a Chef Server for a Splunk Enterprise (server)
+node of type cluster master, that is with `splunk_shclustering_enable:true` and
+the same `splunk_shclustering_label` in the same `chef_environment` and
+use that server's IP when building the list of `shcluster_members`.
+
+The search head cluster configuration is deployed as a custom Splunk app that
+is written to `etc/apps/0_autogen_shcluster_config` to take advantage of Splunk's
+built in config layering. All nodes with `splunk_shclustering_enable:true` will
+receive this app.
+
+On the first Chef run on a node with `splunk_shclustering_mode:captain`, this recipe
+will build and execute the Splunk command to bootstrap the search head cluster and
+initiate the captain election process.
+
+In addition to using this recipe for configuring the search head cluster members, you
+will also have to manually configure a stand-alone search head instance to serve as the 
```

Comment:
Strictly speaking, it doesn't need to be standalone...it could be co-located with a deployment server or a cluster master. As mentioned in another comment, how about supporting this by adding a conditional logic in the same recipe via say a `deployer` mode?